opencl/btx_clinterval_callbacks.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <iostream>
+
+#include "cl.h.include"
+
+#include "xprof_utils.hpp"
+
+using hp_command_queue_t = std::tuple<hostname_t, process_id_t, cl_command_queue>;
+using hp_kernel_t = std::tuple<hostname_t, process_id_t, cl_kernel>;
+using hp_event_t = std::tuple<hostname_t, process_id_t, cl_event>;
+
+inline bool clRvalIsError(cl_int &cl_rval) {
+  return (cl_rval != CL_SUCCESS) && (cl_rval != CL_DEVICE_NOT_AVAILABLE);
+}
```

Comment:
why in the `hpp`? doesn't seem to be templated

Reply:
ya can be moved to cpp as static inline.